app/views/questions/new.html.erb

```diff
@@ -5,14 +5,35 @@
         <div class="box">
           <h2 class="title-no-margin center">New Question</h2>
           <br>
-            <%= form_for @question do |f| %>
+            <%= form_for(@question) do |f| %>
               <div class="form-group">
-                <%= f.text_field :text, placeholder: 'Question', class: 'form-control', autofocus: true %>
+                <%= f.text_field :text, placeholder: 'Question',  class: 'form-control', autofocus: true %>
               </div>
               <div class="form-group">
-                <%= f.label :qset_id, 'Qset', {class: 'control-label'} %>
-                <%= f.collection_select(:qset_id, Qset.find(current_user.org).descendants, :id, :name, {}, {class: 'form-control'}) %>
+                <button type="button" class="btn btn-primary btn-theme btn-generate-q" data-toggle="modal" data-target="#modal-launch">
+                  Choose Qset
+                </button>
+                <% if !params[:qset].nil? %>
+                  <span class="qset-display">
+                    <%= @qsets.find(params[:qset]).name %>
+                  </span>
+                <% elsif !cookies[:new_question_qset_id].nil? %>
+                  <span class="qset-display">
+                    <%= @qsets.find(cookies[:new_question_qset_id]).name %>
+                  </span>
+                <% else %>
+                  <span class="qset-display no-qset-chosen">No question set chosen</span>
+                <% end %>
+                </span>
+                <!-- hidden input field to pass in qset_id into params on submit -->
+                <% if !params[:qset].nil? %>
+                  <input type="text" name="question[qset_id]" id="qset-id-input" class="hidden" value="<%= @qsets.find(params[:qset]).id %>">
```

Comment:
If the `qset` param is just the ID, isn't 
```
@qsets.find(params[:qset]).id
```
equivalent to 
```
params[:qset]
```
...with one less DB access?